Tests/StarknetTests/Providers/ProviderTests.swift

```diff
@@ -396,10 +401,35 @@ final class ProviderTests: XCTestCase {
             classHash: deployAccountTx.classHash
         )
 
-        let simulations2 = try await provider.simulateTransactions([invokeWithoutSignature, deployAccountWithoutSignature], at: .tag(.pending), simulationFlags: [.skipValidate])
+        let simulations2 = try await provider.simulateTransactions([invokeWithoutSignature, deployAccountWithoutSignature], at: .tag(.pending), simulationFlags: [.skipValidate]).send()
 
         XCTAssertEqual(simulations2.count, 2)
         XCTAssertTrue(simulations2[0].transactionTrace is StarknetInvokeTransactionTrace)
         XCTAssertTrue(simulations2[1].transactionTrace is StarknetDeployAccountTransactionTrace)
     }
+
+    func testBatchGetTransactionByHash() async throws {
+        let previousResult = try await batchProvider.getTransactionBy(blockId: .tag(.latest), index: 0).send()
+
+        let transactionsResponse = try await batchProvider.batchRequests(requests:
+            provider.getTransactionBy(hash: previousResult.transaction.hash!),
+            provider.getTransactionBy(hash: "0x123")).send()
+
+        XCTAssertEqual(transactionsResponse.count, 2)
+        XCTAssertEqual(try transactionsResponse[0].get().transaction.hash, previousResult.transaction.hash)
+
+        do {
+            let _ = try transactionsResponse[1].get().transaction.hash
+            XCTFail("Fetching transaction with nonexistent hash should fail")
+        } catch let error as StarknetProviderError {
+            switch error {
+            case let .jsonRpcError(_, message, _):
+                XCTAssertEqual(message, "Transaction hash not found", "Unexpected error message received")
+            default:
+                XCTFail("Expected JsonRpcError but received \(error)")
+            }
+        } catch {
+            XCTFail("Error was not a StarknetProviderError. Received error type: \(type(of: error))")
+        }
```

Comment:
I don't really like the fact that there is double nesting here, but there isn't any other way to access `jsonRpcError` message.